src/utils.js

```diff
@@ -96,13 +96,12 @@ export function featureBagging(trainingSet, n, replacement, seed) {
  * @ignore
  * @param {Matrix} X
  * @param {Array} indexes
- * @return {Matrix} toRet - the new X matrix.
+ * @return {MatrixTransposeView} toRet - the new X matrix.
  */
 export function retrieveFeatures(X, indexes) {
-    var toRet = new Matrix(X.rows, indexes.length);
+    var toRet = new Matrix(indexes.length, X.rows);
     for (var i = 0; i < indexes.length; ++i) {
-        toRet.setColumn(i, X.getColumn(indexes[i]));
+        toRet[i] = X.getColumn(indexes[i]);
```

Comment:
I prefer the setters and getters, they work with views and the performance is the same